### PR TITLE
Fix dependabot automerge

### PR DESCRIPTION
**Pull request by zaneselvans**
Apply the same post-workflow `merge-me-action` to the dependabot PRs as we use for the pre-commit-ci merges, since the dependabot permissions within the `tox-pytest` workflow seem to be insufficient.

### CODE REVIEW

File: .github/workflows/bot-auto-merge.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Auto-merge passing dependabot PRs
+        if: ${{ github.event.workflow_run.conclusion == 'success' }}
+        uses: ridedott/merge-me-action@v2
+        with:
+          # For clarity only. dependabot is default.
+          GITHUB_LOGIN: dependabot
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          ENABLED_FOR_MANUAL_CHANGES: "true"
```
Comment (bendnorman) [inline .github/workflows/bot-auto-merge.yml:21]: So ` ridedott/merge-me-action@v2` only merges PRs created by the bot in `GITHUB_LOGIN`, correct?

**Review by bendnorman — APPROVED** [review]
Looks good. Left a couple of questions.

File: .github/workflows/bot-auto-merge.yml (hunk 1 above)
Comment (zaneselvans) [reply .github/workflows/bot-auto-merge.yml:21]: Yes that's right. This step will only affect `dependabot` PRs. And the other one will only affec `pre-commit-ci` PRs.

File: .github/workflows/bot-auto-merge.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      - name: Auto-merge passing pre-commit-ci PRs
+        if: ${{ github.event.workflow_run.conclusion == 'success' }}
+        uses: ridedott/merge-me-action@v2
+        with:
+          GITHUB_LOGIN: pre-commit-ci
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          ENABLED_FOR_MANUAL_CHANGES: "true"
```
[the remark this answers is not in the document]
Comment (zaneselvans) [reply .github/workflows/bot-auto-merge.yml:28]: Setting this to `true` means that if the CI fails for some reason, and then we go in and fix it, and push the changes to e.g. the dependabot branch, if the CI then passes, the merge will proceed automatically without needing to be manually merged. It seemed convenient, but I could go either way if you'd prefer to remove it.

[the remark this answers is not in the document]
Comment (bendnorman) [reply .github/workflows/bot-auto-merge.yml:28]: Got it, thank you. Seems fine for the PR to be automatically merged in that case. Merge away!

**Pull request merged**